cmd/kube-scheduler/app/server.go

```diff
@@ -79,16 +79,17 @@ suitable Node. Multiple different schedulers may be used within a cluster;
 kube-scheduler is the reference implementation.
 See [scheduling](https://kubernetes.io/docs/concepts/scheduling-eviction/)
 for more information about scheduling and the kube-scheduler component.`,
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			if err := opts.Complete(&namedFlagSets); err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
+				return err
 			}
 			if err := runCommand(cmd, opts, registryOptions...); err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
+				return err
 			}
+			return nil
 		},
+		SilenceErrors: true,
```

Comment:
I'm going to replace this with a more generic solution that'll work for most of the Kubernetes commands in https://github.com/kubernetes/kubernetes/pull/105076.

See https://github.com/kubernetes/kubernetes/pull/105076#discussion_r718247749

It would be good to get some additional eyes on that. Perhaps someone can even try out the PR?

I ran tests manually with kube-scheduler as example and I think it still works as you intended here.